core/src/main/scala/org/http4s/Message.scala

```diff
@@ -201,6 +205,11 @@ case class Request(
 
   override def toString: String =
     s"""Request(method=$method, uri=$uri, headers=${headers}"""
+
+  /** A request is idempotent iff its method is idempotent and its body is pure.
```

Comment:
@rossabaker, I had to look up iff, I thought it was a spelling mistake but that didn't seem characteristic of you. Maybe we should just spell out 'if and only if' for those who might not know what `iff` means? Or maybe I'm just really late to the party...